include/deal.II/matrix_free/tools.h

```diff
@@ -650,6 +699,13 @@ namespace MatrixFreeTools
     const unsigned int                                              quad_no,
     const unsigned int first_selected_component)
   {
+    std::unique_ptr<AffineConstraints<typename MatrixType::value_type>>
+                                                              constraints_for_matrix;
+    const AffineConstraints<typename MatrixType::value_type> &constraints =
+      internal::create_new_affine_constraints_if_needed(matrix,
+                                                        constraints_in,
+                                                        constraints_for_matrix);
```

Comment:
If the underlying type of `AffineConstraints` object and of the matrix do not match, I create a copy of the constraints.